src/forcegraph-kapsule.js

```diff
@@ -237,9 +237,24 @@ export default Kapsule({
 
           const pos = isD3Sim ? node : state.layout.getNodePosition(node[state.nodeId]);
 
-          obj.position.x = pos.x;
-          obj.position.y = pos.y || 0;
-          obj.position.z = pos.z || 0;
+          if (obj.isInstancedMesh) {
+            // get matrix
+            obj.getMatrixAt(node.index, obj.matrix);
```

Comment:
This approach assumes that the instance index is stored in an `index` attribute of the data object. This seems too coupled with a specific implementation of the instanced mesh. Additionally you have the problem that in this manner you will add an `InstancedMesh` to the scene for every node, where as you ideally just want to add it once to the scene.

Reply:
Ack! I understand, I will try to find a workaround to this and make the approach more agnostic. To leave more context and to make sure I'm not missing anything out, I'm trying to use an `InstancedMesh` for the nodes in order to save performance - did a couple tests and I gain fps implementing the following. 

```js
const nodeMesh = new InstancedMesh(
  new BoxGeometry(4, 4, 4),
  new MeshBasicMaterial({
  color: 0xffffff,
  opacity: 0.5,
  transparent: true,
  }),
  graph.graphData().nodes.length,
);

const object3D = new Object3D();

// set the node mesh as the object for each node
graph.nodeThreeObject((node) => {
  // set the scale of the node
  object3D.scale.set(1.5, 1.5, 1.5);
  
  // update the matrix of the object
  object3D.updateMatrix();
  
  // set matrix of the instanced mesh to the node's matrix
  nodeMesh.setMatrixAt(node?.id ?? 0, object3D.matrix);
  
  nodeMesh.instanceMatrix.needsUpdate = true;
  
  return nodeMesh;
});
```
Is there perhaps a better way to implement this?

Reply:
I've just added a method to this module that will probably help you achieve what you're looking for.

It's called `nodePositionUpdate`. With it, you can override the internal behaviour of how positions get assigned to objects, so you can put your logic from this PR inside that method. Just make sure to return `true` from that function to instruct the module to not run its own position updater.

See more in [the docs](https://github.com/vasturiano/three-forcegraph#api-reference).

Reply:
This worked perfectly for our use case, thanks! I did a follow up PR (#35) that adds types for this. Closing the PR 🎊